medicines/doc-index-updater/src/models.rs

```diff
@@ -29,11 +30,18 @@ impl FromStr for JobStatus {
         match s {
             "Accepted" => Ok(JobStatus::Accepted),
             "Done" => Ok(JobStatus::Done),
-            "Error" => Ok(JobStatus::Error {
-                message: "Error status".to_owned(),
-                code: "0x0".to_owned(),
-            }),
-            e => Err(format!("Status unknown: {}", e)),
+            status => {
+                // If this message is in the format "Error(error code: error message)",
+                // reconstruct it into JobStatus::Error.
+                let error_re = Regex::new(r"^Error\((?P<code>[^:]*): (?P<message>.*)\)$").unwrap();
+                match error_re.captures(status) {
+                    Some(capture) => Ok(JobStatus::Error {
+                        message: capture.name("message").unwrap().as_str().to_string(),
```

Comment:
These go through quite a journey! A result -> str -> String!

Reply:
It's nuts. The captures have an `as_str` but no `to_string`.